torchaudio/datasets/quesst14.py

```diff
@@ -0,0 +1,116 @@
+import os
+import re
+from pathlib import Path
+from typing import Tuple, Union, Optional
+
+import torch
+from torch.hub import download_url_to_file
+from torch.utils.data import Dataset
+from torchaudio.datasets.utils import extract_archive
+from torchaudio.sox_effects import apply_effects_file
+
+
+URL = "https://speech.fit.vutbr.cz/files/quesst14Database.tgz"
+_CHECKSUM = "4f869e06bc066bbe9c5dde31dbd3909a0870d70291110ebbb38878dcbc2fc5e4"
+_LANGUAGES = [
+    "albanian",
+    "basque",
+    "czech",
+    "nnenglish",
+    "romanian",
+    "slovak",
+]
+
+
+class QUESST14(Dataset):
+    """Create QUESST14 Dataset
+
+    Args:
+        root (str or Path): Root directory where the dataset's top level directory is found
+        subset (str): The subset to use. Options: [``dev``, ``eval``]
+        language (str, optional): Language to get dataset for.
+            Options: [None, ``albanian``, ``basque``, ``czech``, `nnenglish``, ``romanian``, ``slovak``]
+        folder_in_archive (str, optional): The top-level directory of the dataset. default: (``"quesst14Database"``)
+        subset (str or None, optional): subset of the dataset to use. Options: [None, "dev", "eval"].
+            None indicates the whole dataset, including dev and eval.
+        download (bool, optional): Whether to download the dataset if it is not found at root path.
+            (default: ``False``)
+    """
+
+    def __init__(
+        self,
+        root: Union[str, Path],
```

Comment:
I am wondering if we can use a default path for the download location, similar to pretained model. Sure it will be different from the previous datasets, but I often find that it's more convenient that way. Thoughts?

Reply:
The default path will be tricky if we consider Windows (`C:\`), Linux and MacOS (`/usr/datasets/`).

Reply:
this seems out of scope for this PR, but I agree that it can be good for managing consistent dataset paths and allow us to better maintain cached dataset artifacts, and would not negatively affect users as they can still input their own root path if they have preferences. do you have a suggestion on the path location?